TACC-Cloud/hazmapper
# task/WG-366: update react frontend to use tapisv3

## Overview: ##

This PR updates react client to use tapisv3:

* uses new auth flow.
* updates requests to tapis

Also in this PR is the following:
* adds on click behavior to navigate to map project (to ease testing)
* fixes one warning in tests

## PR Status: ##


## Related Jira tickets: ##

* [WG-366](https://tacc-main.atlassian.net/browse/WG-366)

## Testing Steps: ##

1. ** do not use dev geoapi as backend as still down; use Staging or Production in `react/src/secret_local.ts`  **
2. Run app and look at project listing.
3. Click on an existing project and see that map is populated with features as expected


## Notes ##

- [x] add check for when token is about to expire
- [x] add jira issue for failing on test warnings (request is made in one of the tests) https://tacc-main.atlassian.net/browse/WG-368
- [x] add click on project route and navigate to project (to ease testing)
